package/MDAnalysis/auxiliary/EDR.py

```diff
@@ -0,0 +1,357 @@
+
+The EDRReader can be used to add auxiliary data from EDR files to
+trajectories::
+
+    # Create a Universe
+    In [1]: u = mda.Universe(AUX_EDR_TPR, AUX_EDR_XTC)
+    # Open the EDR file with an EDRReader
+    In [2]: aux = mda.auxiliary.EDR.EDRReader(AUX_EDR)
+    # See which terms are present in the file
+    In [3]: aux.terms
+    Out[3]: ['Time', 'Bond', 'Angle', ...]
+    # add the Potential data under the name epot
+    In [4]: u.trajectory.add_auxiliary("epot", aux, "Potential")
+    # add the Angle energy without stating an `auxterm`
+    In [5]: u.trajectory.add_auxiliary("Angle", aux)
+    # access the value of Potential and Angle for a timestep
+    In [6]: u.trajectory.ts.aux.epot
+    Out[6]: -525164.0625
+    In [7]: u.trajectory.ts.aux.Angle
+    Out[7]: 3764.52734375
+    # dictionary style accession also works
+    In [8]: u.trajectory.ts.aux["epot"]
+    Out[8]: -525164.0625
+
+
+Note: Some `auxterms` have spaces in their names. Unless an `auxname` without a
+space is provided, these terms will not be accessible via the attribute syntax.
+Only the dictionary syntax will work in that case.
+
+
+Adding multiple or all terms at the same time is possible as such::
+
+    # add list of values
+    u.trajectory.add_auxiliary(["Bond", "Angle"], aux)
+    # add list of values, provide an `auxterm`
+    u.trajectory.add_auxiliary(["epot", "ekin"], aux, ["Potential",
+                                                       "Kinetic En."]
+    # add all values
+    u.trajectory.add_auxiliary("*", aux)
+
+
+
+The EDRReader can also provide the data independently of trajectories. This is
+useful, for example, for plotting. The data for all terms, a list of terms, or
+a single term can be returned in dictionary form. "Time" is always returned in
+this dictionary to make plotting easier::
+
+    # extract temperature data from EDR file
+    temp = aux.return_data("Temperature")
+    plt.plot(temp["Time"], temp["Temperature"])
+
+    # extract list of terms:
+    some_terms = aux.return_data(["Potential", "Kinetic En.", "Box-X"])
+
+    # extract all terms:
+    all_terms = aux.return_data()
+    # alternatively
+    all_terms = aux.return_data("*")
+
+.. _EDR: https://manual.gromacs.org/current/reference-manual/file-formats.html#edr
+.. _XDR: https://datatracker.ietf.org/doc/html/rfc1014
+.. _pyedr: https://github.com/mdanalysis/panedr
+.. _GROMACS: https://github.com/gromacs/gromacs/blob/main/src/gromacs/fileio/enxio.cpp
+
+
+.. autoclass:: EDRReader
+   :members:
+
+
+"""
+from pathlib import Path
+from . import base
+import pyedr
+import numpy as np
+from typing import Optional, Union, Dict, List
+
+
+class EDRStep(base.AuxStep):
+    """:class:`AuxStep` class for .edr file format.
+
+    Extends the base AuxStep class to allow selection of time and
+    data-of-interest fields (by column index) from the full set of data read
+    each step.
+
+    Parameters
+    ----------
+    time_selector : int | None, optional
```

Comment:
```suggestion
    time_selector : str, optional
```
From the `__init__` signature?

Reply:
Oh, yes... Need to change the docstring here in general still, thanks!